scimath/Mathematics/ClassicalStatistics.tcc

```diff
@@ -2352,10 +2403,10 @@ void ClassicalStatistics<CASA_STATP>::_makeBins(
 #define _minMaxCode \
     if (! mymin.null()) { \
         if (*datum < *mymin) { \
-            mymin = new AccumType(*datum); \
+            *mymin = *datum; \
         } \
         else if (*datum > *mymax) { \
```

Comment:
hm being shared pointers breaks the compilers ability to use the branchless instructions and vectorize
why are they shared pointers?
If they have to be a double and float specialization is probably worthwhile
also pull out the null check can be pulled out of the loops and replaced with first entry assignment.